lib/actions.ex

```diff
@@ -224,11 +427,89 @@ defmodule EctoShorts.Actions do
     field :: atom
   ) :: term
   def aggregate(schema, params, aggregate, field, opts \\ []) do
-    Repo.call(:aggregate, [
+    repo(opts).aggregate(
       CommonFilters.convert_params_to_filter(schema, params),
       aggregate,
       field,
       opts
-    ])
+    )
+  end
+
+  @doc """
+  Accepts a list of schemas and attempts to find them in the DB. Any missing Schemas will be created.
+  Can also accept a keyword options list.
+
+  ## Options
+    * `:repo` - A module that uses the Ecto.Repo Module.
+    * `:replica` - If you don't want to perform any reads against your Primary, you can specify a replica to read from.
+
+  ## Examples
+    iex> {:ok, records} = EctoSchemas.Actions.find_or_create_many(EctoSchemas.Accounts.User, [%{name: "foo"}, %{name: "bar}])
+    iex> length(records) === 2
+  """
+  @spec find_or_create_many(
+    Ecto.Schema.t(),
+    list(map),
+    opts :: Keyword.t
+  ) :: {:ok, list(Ecto.Schema.t())} | {:error, list(Ecto.Changeset.t())}
+  def find_or_create_many(schema, param_list, opts) do
+    {create_params, found_results} = find_many(schema, param_list, opts)
+
+    schema
+    |> multi_insert(param_list, create_params)
+    |> repo(opts).transaction()
+    |> case do
+      {:ok, created_map} -> {:ok, merge_found(created_map, found_results)}
+      error -> error
+    end
+  end
+
+
+  defp find_many(schema, param_list, opts) do
+    param_list
+    |> Enum.map(fn params ->
+      case find(schema, params, opts) do
+        {:ok, result} -> result
+        _ -> nil
+      end
+    end)
+    |> Enum.with_index()
+    |> Enum.split_with(fn {result, _index} -> is_nil(result) end)
+  end
+
+  defp multi_insert(schema, param_list, create_params) do
+    Enum.reduce(create_params, Ecto.Multi.new(), fn {nil, i}, multi ->
+      Ecto.Multi.insert(multi, i, fn _ ->
+        param_list
+        |> Enum.at(i)
+        |> schema.create_changeset
+      end)
+    end)
+  end
+
+  defp merge_found(created_map, found_results) do
+    created_map
+    |> Enum.map(fn {index, result} -> {result, index} end)
+    |> Kernel.++(found_results)
+    |> Enum.sort(&(elem(&1, 1) >= elem(&2, 1)))
+    |> Enum.map(&elem(&1, 0))
+  end
+
+  defp reduce_status_tuples(status_tuples) do
+    {status, res} =
+      Enum.reduce(status_tuples, {:ok, []}, fn
+        {:ok, _}, {:error, _} = e -> e
+        {:ok, record}, {:ok, acc} -> {:ok, [record | acc]}
+        {:error, error}, {:ok, _} -> {:error, [error]}
+        {:error, e}, {:error, error_acc} -> {:error, [e | error_acc]}
+      end)
+
+    {status, Enum.reverse(res)}
+  end
+
+  defp repo(opts) do
+    @default_opts
```

Comment:
Can we add a with statement here and raise if it's nil? We can provide a clear message about setting repo that would make mistakes clear! Since right now you'd get a nasty function on nil undefined error